src/pages/property/[uid].js

```diff
@@ -52,125 +59,174 @@ function groupArr(data, n) {
 		}, []);
 }
 
-export default function ComplexGrid({ property }) {
-	const callme = () => {
-		console.log('hi');
+export default function PropertyDetails({ property, users }) {
+	const [user, loading] = useAuthUser();
+	const [like, setLike] = useState(false);
+
+	useEffect(() => {
+		if (user) {
+			getPropertyListFromUserID(user.uid, property.uid)
+				.then((res) => {
+					setLike(res.includes(property.uid));
+				})
+				.catch((err) => {
+					console.log('err', err);
+				});
+		}
+	}, [user, property.uid]);
+
+	const handleLike = (e) => {
+		setLike(!like);
+		if (like) {
+			deleteUserPropertyMapping(user.uid, property.uid);
+		} else {
+			addUserPropertyPreference(user.uid, property.uid);
+		}
```

Comment:
same note as earlier, please also update this function. 

if this function is duplicated should it be defined somewhere else and both can import? i don't know where that'd be so i'm okay leaving it but definitely not the greatest practice

Reply:
In this case I can't find an easy way to arrange for the best practice (more maintainable) since this function requires other three functions and one variable. I will just leave it for now.